data-prepper-plugins/mutate-event-processors/src/test/java/com/amazon/dataprepper/plugins/processor/mutateevent/AddEntryProcessorTests.java

```diff
@@ -0,0 +1,208 @@
+        assertThat(editedRecords.get(0).getData().get("newMessage", Object.class), equalTo(true));
+    }
+
+    @Test
+    public void testStringAddProcessorTests() {
+        when(mockConfig.getEntries()).thenReturn(createListOfEntries(createEntry("newMessage", "string", false)));
+
+        final AddEntryProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent("thisisamessage");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("thisisamessage"));
+        assertThat(editedRecords.get(0).getData().containsKey("newMessage"), is(true));
+        assertThat(editedRecords.get(0).getData().get("newMessage", Object.class), equalTo("string"));
+    }
+
+    @Test
+    public void testNullAddProcessorTests() {
+        when(mockConfig.getEntries()).thenReturn(createListOfEntries(createEntry("newMessage", null, false)));
+
+        final AddEntryProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent("thisisamessage");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("thisisamessage"));
+        assertThat(editedRecords.get(0).getData().containsKey("newMessage"), is(true));
+        assertThat(editedRecords.get(0).getData().get("newMessage", Object.class), equalTo(null));
+    }
+
+    @Test
+    public void testFloatAddProcessorTests() {
+        when(mockConfig.getEntries()).thenReturn(createListOfEntries(createEntry("newMessage", 1.2, false)));
+
+        final AddEntryProcessor processor = createObjectUnderTest();
+        final Record<Event> record = getEvent("thisisamessage");
+        final List<Record<Event>> editedRecords = (List<Record<Event>>) processor.doExecute(Collections.singletonList(record));
+
+        assertThat(editedRecords.get(0).getData().containsKey("message"), is(true));
+        assertThat(editedRecords.get(0).getData().get("message", Object.class), equalTo("thisisamessage"));
+        assertThat(editedRecords.get(0).getData().containsKey("newMessage"), is(true));
+        assertThat(editedRecords.get(0).getData().get("newMessage", Object.class), equalTo(1.2));
+    }
+
+    private AddEntryProcessor createObjectUnderTest() {
+        return new AddEntryProcessor(pluginMetrics, mockConfig);
+    }
+
+    private AddEntryProcessorConfig.Entry createEntry(final String key, final Object value, final boolean overwriteIfKeyExists) {
+        return new AddEntryProcessorConfig.Entry(key, value, overwriteIfKeyExists);
```

Comment:
You could probably mock this object and set behaviors on it. Then you could remove the constructors from the `Entry` classes.

But, I'm fine with it either way.

Reply:
Oh yeah, I hadn't thought of that. I'll do that method next time!